### PR TITLE
Reimplement getPortByIndex in SingularityTask

### PR DESCRIPTION
**Pull request by ssalinas**
This was removed in the removal of the mesos dep from SingularityBase. Adding it back in for convenience.

### CODE REVIEW

File: SingularityBase/src/main/java/com/hubspot/mesos/protos/MesosRangesObject.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    this.range = range;
+  }
+
+  public List<MesosRangeObject> getRange() {
+    return range.or(Collections.emptyList());
+  }
+
+  @JsonIgnore // to mimic mesos
+  public List<MesosRangeObject> getRangesList() {
+    return getRange();
```
Comment (darcatron) [inline SingularityBase/src/main/java/com/hubspot/mesos/protos/MesosRangesObject.java]: not sure I understand the reason for this method. doesn't seem to do anything different than `getRange()`

**Comment by baconmania** [comment]
🚢

File: SingularityBase/src/main/java/com/hubspot/mesos/protos/MesosRangesObject.java (hunk 1 above)
Comment (baconmania) [reply SingularityBase/src/main/java/com/hubspot/mesos/protos/MesosRangesObject.java]: It's to mimic the convention used by the Mesos API, with stuff like [`getPortsList()`](http://mesos.apache.org/api/latest/java/org/apache/mesos/Protos.PortsOrBuilder.html#getPortsList--).

Comment (ssalinas) [reply SingularityBase/src/main/java/com/hubspot/mesos/protos/MesosRangesObject.java]: correct, added with the aim of not breaking binary compatibility for those who were using the old mesos protos objects

**Pull request merged**